articles/153_roslaunch_remote_launch.md

```diff
@@ -0,0 +1,185 @@
+| log\_cmd             | boolean | If `True`, prints the final cmd before executing the process, which is useful for debugging when substitutions are involved. |
+| on\_exit             | List[LaunchDescriptionEntity] | List of actions to execute upon process exit.|
+| persistent\_connection | boolean | Whether the LaunchService should maintain a persistent connection to the ssh instance |
+| machine              | `launch.descriptions.SSHMachine` | The machine on which to execute the process |
+##### Methods
+| Name               | Type | Description                                                  |
+| ------------------ | --- | ------------------------------------------------------------ |
+| get\_sub\_entities | List[LaunchDescriptionEntity] | Override. If on\_exit was provided in the constructor, returns that; otherwise returns an empty list. |
+| execute            | Optional[List[LaunchDescriptionEntity]] | Override. Establishes event handlers for process execution, passes the execution context to the process definition for substitution expansion, then uses `osrf_pycommon.process_utils.async_execute_process` to launch the defined process. |
+| get\_asyncio\_future | Optional[asyncio.Future] | Override. Return an asyncio Future, used to let the launch system know when we're done. |
+
+
+#### launch.actions.ExecuteRemoteSSH
+Inherits from `launch.actions.ExecuteRemote` and manages the ssh connection
+##### Members
+Inherited
+##### Methods
+| Name | Type | Description |
+|---|---|---|
+| open\_ssh\_connection | SSHConnection | Object for interacting with the SSH connection |
+| exec\_remote\_command | void | Executes a command on the remote machine
+
+#### launch.substitutions.RemoteSubstitution
+Uses a connection to a machine to run a command remotely and retuns the result as a string
+##### Members
+##### Constructor
+| Argument | Type | Description |
+|---|---|---|
+| command | Text | The command to run on the remote machine |
+| machine | `launch.descriptions.Machine` | An object decribing the machine that the command should be run on. |
+##### Methods
+| Name | Type | Description |
+|---|---|---|
+| describe | Text | Override method. Returns a description of this substitution as a string. |
+| perform  | Text | Override method. Perform the substitution, given the launch context, and return it as a string. |
+
+
+## ROS
+
+### Nodes
+#### launch\_ros.LaunchServiceNode
+This node runs on each machine in a system is is repsonsible for launching and managing remote processes.
+#### Params
+| Name | Type | Description |
+|---|---|---|
+| allowable\_processes | List[Executable] | List of executables the node is allowed to launch |
+##### Topics
+###### Sent
+- heartbeat
+###### Received
+##### Services
+- query\_status
+- start\_process
+- stop\_process
+- shutdown
+
+### Messages
+#### Heartbeat
```

Comment:
Do all LaunchServiceNodes use a unique topic, or do all publish on the same topic (in which case would need to put machine/service ID in the messages)

Reply:
Maybe this should be configurable, "heartbeat_topic", "process_status_topic"